target/linux/ath79/dts/qca9563_tplink_cpe710-v1.dts

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca956x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "TP-Link CPE710 v1";
+	compatible = "tplink,cpe710-v1", "qca,qca9563";
+
+	aliases {
+		label-mac-device = &eth0;
+		led-boot = &led_lan;
+		led-failsafe = &led_lan;
+		led-upgrade = &led_lan;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_lan: lan {
+			label = "blue:lan";
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "blue:wlan5g";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 2 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "u-boot";
+				reg = <0x000000 0x040000>;
+				read-only;
+			};
+
+			partition@40000 {
+				label = "u-boot-env";
```

Comment:
Does this work? This is the only tp-link ath79 device with such a partition defined. Does it need the read-only removed to write to it?

Reply:
Yes it works.
I personally don't see why we would need to write to this partition. 
 I have it defined this way to match the OEM way of doing it 
The Entry from the OEM boot log is
[    0.980000] m25p80 spi0.0: found gd25q128, expected m25p80
[    0.990000] m25p80 spi0.0: gd25q128 (16384 Kbytes)
[    0.990000] 8 cmdlinepart partitions found on MTD device spi0.0
[    1.000000] Creating 8 MTD partitions on "spi0.0":
[    1.010000] 0x000000000000-0x000000040000 : "u-boot"
**_[    1.010000] 0x000000040000-0x000000050000 : "u-boot-env"_**
[    1.020000] 0x000000050000-0x000000070000 : "info"
.....

Reply:
Re removing the read-only flag, see the end of https://openwrt.org/docs/techref/bootloader/uboot.config
I guess getting it properly configured could be done in seperate PRs.

Reply:
if there is a good reason to make uboot-env writable, it would be handled in this PR

I don't see the need, but I usually do this for boards I add whether it's needed or not

Reply:
u-boot-env should only be writable if there is a reason for it (typically uboot-envtools entry etc.).

Reply:
@jwmullally I have removed the read-only from u-boot-env partition. Please test with your device.